management/clipper_manager.py

```diff
@@ -467,28 +469,64 @@ def centered_predict(inputs):
         if not os.path.exists(serialization_dir):
             os.makedirs(serialization_dir)
 
-        # Write out function serialization
-        func_file_path = "{dir}/{predict_fname}".format(
-            dir=serialization_dir, predict_fname=predict_fname)
-        with open(func_file_path, "w") as serialized_function_file:
-            serialized_function_file.write(serialized_prediction_function)
-        print("Serialized and supplied predict function")
-
         # Export Anaconda environment
         subprocess.call(
             "PIP_FORMAT=legacy conda env export >> {environment_fname}".format(
                 environment_fname=environment_fname),
             shell=True)
 
+        # Confirm that packages installed through conda are solvable
+        if not (self._conda_env_solvable(environment_fname, os.getcwd())):
+            return
+
         # Give container environment details
         shutil.copy(environment_fname, serialization_dir)
         print("Supplied environment details")
 
+        # Write out function serialization
+        func_file_path = "{dir}/{predict_fname}".format(
+            dir=serialization_dir, predict_fname=predict_fname)
+        with open(func_file_path, "w") as serialized_function_file:
+            serialized_function_file.write(serialized_prediction_function)
+        print("Serialized and supplied predict function")
+
         # Deploy function
         return self.deploy_model(name, version, serialization_dir,
                                  default_python_container, labels, input_type,
                                  num_containers)
 
+    def _conda_env_solvable(self, environment_fname, directory):
+        """Returns true if the provided conda environment is compatible with the container os.
+
+        If packages listed in specified conda environment file have conflicting dependencies,
+        this function will warn the user and return False. If packages don't exist in the
+        container's conda channel, this function will warn the user and remove those packages.
+
+        Parameters
+        ----------
+        environment_fname : str
+            The file name of the exported conda environment file
+        directory : str
+            The path to the diretory containing the environment file
+
+        Returns
+        -------
+        bool
+            Returns True if the (possibly modified) environment file is compatible with conda
+            on the container os. Otherwise returns False.
+        """
+
+        child = subprocess.Popen(
+            "source deactivate && python check_env.py {environment_fname} {directory} {platform}".
+            format(
+                environment_fname=environment_fname,
+                directory=directory,
+                platform=CONTAINER_CONDA_PLATFORM),
+            stderr=subprocess.PIPE,
+            shell=True)
+        child.communicate()
+        return child.returncode
```

Comment:
Oh also. Remember that when you're printing things in a Python script, it's being printed to stdout, so maybe you want to capture that as well.